src/window/action_popup.c

```diff
@@ -156,28 +147,32 @@ static void handle_input(const mouse *m, const hotkeys *h)
         window_go_back();
     }
 
-    if (generic_buttons_handle_mouse(mouse_in_dialog(m), 80, 40, buttons, data.show_cancel_button ? 4 : 3, &data.focus_button_id)) {
+    if (generic_buttons_handle_mouse(mouse_in_dialog(m), 80, 40, buttons, 5, &data.focus_button_id)) {
         return;
     }
-
 }
 
 static void button_select_option(int option, int param2)
 {
-    if (option == 1 || option == 2 || option == 3) {
-        data.close_func(option);
-        window_go_back();
-    } else if (!option) {
-        data.close_func(option);
-        window_go_back();
+    if (option && option != CONFIRM_BUTTON) {
+        data.selected_policy = option;
```

Comment:
Put all the texts on `draw_background` and call `window_request_refresh()` here. It will refresh the full contents of the window and not only fix your text issue, but also be more performant by only redrawing when actually needed.

Reply:
Thx it was my original code but without request_refresh...